src/diffusers/models/embeddings.py

```diff
@@ -249,6 +249,30 @@ def forward(self, x):
         return out
 
 
+class PositionalEmbedding(nn.Module):
+    """Apply positional information to a sequence of embeddings.
+
+    Takes in a sequence of embeddings with shape (batch_size, seq_length, embed_dim) and adds positional embeddings to
+    them
+
+    Args:
+        embed_dim: (int): Dimension of the positional embedding.
+        max_seq_length: Maximum sequence length to apply positional embeddings
+
+    """
+
+    def __init__(self, embed_dim: int, max_seq_length: int = 32):
+        super().__init__()
+        pe = get_1d_sincos_pos_embed_from_grid(embed_dim, torch.arange(32))
+        pe = torch.tensor(pe, dtype=torch.float32).unsqueeze(0)
+        self.register_buffer("pe", pe)
```

Comment:
```suggestion
```

Reply:
@patrickvonplaten is there a specific reason to avoid using `self.register_buffer`? It makes changing dtypes and devices much cleaner IMO.

Reply:
Yeah I'd usually just do this in the forward method, but ok to leave it. The reason I don't like to register as a buffer is because it'll save the sinus position embeddings as part of the weights, making them unnecessarily heavy. But ok to leave!